--- a/source/TrampolineBridge/ClosureTrampolineBridge/arm64/closure_bridge_arm64.asm
+++ b/source/TrampolineBridge/ClosureTrampolineBridge/arm64/closure_bridge_arm64.asm
@@ -50,8 +50,13 @@ cdecl(closure_bridge_asm):
 // call convention: x0 = register context, x1 = interceptor entry
 mov x0, sp
 ldr x1, [sp, #(2 * 8 + 2 * 8 + 30 * 8 + 8 * 16)]
+#if defined(__APPLE__)
 adrp TMP_REG_0, cdecl(common_closure_bridge_handler)@PAGE
 add TMP_REG_0, TMP_REG_0, cdecl(common_closure_bridge_handler)@PAGEOFF
+#else
+adrp TMP_REG_0, cdecl(common_closure_bridge_handler)
+add TMP_REG_0, TMP_REG_0, #:lo12:cdecl(common_closure_bridge_handler)
+#endif
 blr TMP_REG_0
 
 // restore stack, saved original sp
